docs/tutorials/TracingCode.adoc

```diff
@@ -0,0 +1,270 @@
+`seedu.address.ui.CommandBox.CommandExecutor`.
+
+.Using the `Search for target by name` feature. `Navigate` > `Symbol`.
+image::Execute.png[width=1011px, height=372px]
+{empty} +
+
+A quick look at the class confirms that this is indeed close to what we're looking for. However, it is just an
+`Interface`. Let's delve further and find the implementation of the interface by using the `Find Usages` feature in
+IntelliJ IDEA.
+
+.`Find Usages` tool window. `Edit` > `Find` > `Find Usages`.
+image::FindUsages.png[width=968px, height=394px]
+{empty} +
+
+Bingo! `MainWindow#executeCommand()` seems to be exactly what we're looking for!
+
+Now let's set the breakpoint. First, double-click the item to reach the corresponding code. Once there, click on the left
+gutter to set a breakpoint, as shown below.
+image:LeftGutter.png[width=1171px, height=154px]
+
+
+[qanda]
+
+
+== Tracing the execution path
+
+Recall from the User Guide that the `edit` command has the format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL]
+[a/ADDRESS] [t/TAG]...` For this tutorial we will be issuing the command `edit 1 n/Alice Yeoh`.
+
+[TIP]
+.Taking notes
+====
+Over the course of the debugging session, you will encounter every major component in the application. Try to jot down
+what happens inside the component and where the execution transfers to another component.
+====
+
+. To start the debugging session, simply `Run` > `Debug `Main`
+. Enter `edit 1 n/Alice Yeoh` into the command box and press `Enter`.
+. The Debugger tool window should show up and look something like this:
++
+image:DebuggerStep1.png[width=1351px,height=485px]
+. Use the `Show execution point` feature to jump to the line of code that we stopped at:
++
+image:ShowExecutionPoint.png[width=525px,height=91px]
+. `CommandResult commandResult = logic.execute(commandText);` is the line that you end up at.
+. We are interested in the `logic.execute(commandText)` portion of that line so let's `Step in` into that method call:
++
+image:StepInto.png[width=525px, height=90px]
+. We end up in `LogicManager#execute()`. Let's take a look at the body of the method and annotate what we can deduce.
++
+.LogicManager#execute()
+[source, java]
+----
+@Override
+public CommandResult execute(String commandText) throws CommandException, ParseException {
+     logger.info("----------------[USER COMMAND][" + commandText + "]"); //Logging, safe to ignore
+
+     CommandResult commandResult;
+     Command command = addressBookParser.parseCommand(commandText); //Parse user input from String to a Command
+     commandResult = command.execute(model); //Executes the Command and stores the result
+
+     try {
+         //We can deduce that the previous line of code modifies model in some way since it's being stored here.
+         storage.saveAddressBook(model.getAddressBook());
+     } catch (IOException ioe) {
+         throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
+     }
+
+     return commandResult;
+ }
+----
```

Comment:
Code overflowing to the next line

Reply:
Anything over 120 overflows when the screen is maximized, fixed all code examples.